### PR TITLE
CosmosDB (DocumentDB) Accounts

### PR DESCRIPTION
**Pull request by tombuildsstuff**
Porting this over from the main repository / https://github.com/hashicorp/terraform/pull/11499

- [x] Import the vendored Azure SDK
- [x] Functionality
  - [x] CRUD
     - [x] Create
     - [x] Read
     - [x] Update
     - [x] Delete
  - [x] Validation
    - [x] Consistency Policy
      - [x] Consistency Level
      - [x] Max Interval in Seconds
      - [x] Max Staleness
    - [x] Failover Policy
      - [x] Priority
    - [x] SKUs
- [x] Renaming
  - [x] DocumentDB -> CosmosDB
  - [x] `azurerm_cosmos_db` -> `azurerm_cosmosdb_account`
- [x] Documentation
- [x] ~[Open a bug about `max_interval_in_seconds` / `max_staleness_prefix` being required, when the docs say otherwise](https://github.com/Azure/azure-rest-api-specs/issues/1324)~
- [x] Acceptance Testing
  - [x] All Supported SKU's
  - [x] Test expanding Geo Replication
  - [x] Validation
  - [x] Import
  - [x] Ensure Passing

### CODE REVIEW

**Comment by tombuildsstuff** [comment]
Tests pass:

```
$ TF_ACC=1 envchain azurerm go test ./azurerm -v -run TestAccAzureRMCosmosDB -timeout 300m

=== RUN   TestAccAzureRMCosmosDBAccount_importBoundedStaleness
--- PASS: TestAccAzureRMCosmosDBAccount_importBoundedStaleness (767.53s)
=== RUN   TestAccAzureRMCosmosDBAccount_importBoundedStalenessComplete
--- PASS: TestAccAzureRMCosmosDBAccount_importBoundedStalenessComplete (761.12s)
=== RUN   TestAccAzureRMCosmosDBAccount_importEventualConsistency
--- PASS: TestAccAzureRMCosmosDBAccount_importEventualConsistency (760.73s)
=== RUN   TestAccAzureRMCosmosDBAccount_importSession
--- PASS: TestAccAzureRMCosmosDBAccount_importSession (742.40s)
=== RUN   TestAccAzureRMCosmosDBAccount_importStrong
--- PASS: TestAccAzureRMCosmosDBAccount_importStrong (764.82s)
=== RUN   TestAccAzureRMCosmosDBAccount_importGeoReplicated
--- PASS: TestAccAzureRMCosmosDBAccount_importGeoReplicated (1606.67s)
=== RUN   TestAccAzureRMCosmosDBAccountName_validation
--- PASS: TestAccAzureRMCosmosDBAccountName_validation (0.00s)
=== RUN   TestAccAzureRMCosmosDBAccount_boundedStaleness
--- PASS: TestAccAzureRMCosmosDBAccount_boundedStaleness (1012.67s)
=== RUN   TestAccAzureRMCosmosDBAccount_boundedStalenessComplete
--- PASS: TestAccAzureRMCosmosDBAccount_boundedStalenessComplete (745.02s)
=== RUN   TestAccAzureRMCosmosDBAccount_eventualConsistency
--- PASS: TestAccAzureRMCosmosDBAccount_eventualConsistency (751.21s)
=== RUN   TestAccAzureRMCosmosDBAccount_session
--- PASS: TestAccAzureRMCosmosDBAccount_session (758.07s)
=== RUN   TestAccAzureRMCosmosDBAccount_strong
--- PASS: TestAccAzureRMCosmosDBAccount_strong (777.53s)
=== RUN   TestAccAzureRMCosmosDBAccount_geoReplicated
--- PASS: TestAccAzureRMCosmosDBAccount_geoReplicated (2330.66s)
```

File: website/docs/r/cosmosdb_account.html.markdown [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+* `ip_range_filter` - (Optional) DocumentDB Firewall Support: This value specifies the set of IP addresses or IP address ranges in CIDR form to be included as the allowed list of client IP's for a given database account. IP addresses/ranges must be comma separated and must not contain any spaces.
+
+* `tags` - (Optional) A mapping of tags to assign to the resource.
+
+`consistency_policy` supports the following:
+
+* `consistency_level` - (Required) The Consistency Level to use for this CosmosDB Account - can be either `BoundedStaleness`, `Eventual`, `Session` or `Strong`.
+* `max_interval_in_seconds` - (Optional) When used with the Bounded Staleness consistency level, this value represents the time amount of staleness (in seconds) tolerated. Accepted range for this value is 1 - 100. Defaults to `5`. Required when `consistency_level` is set to `BoundedStaleness`.
+* `max_staleness` - (Optional) When used with the Bounded Staleness consistency level, this value represents the number of stale requests tolerated. Accepted range for this value is 1 – 2,147,483,647. Defaults to `100`. Required when `consistency_level` is set to `BoundedStaleness`.
```
Comment (tombuildsstuff) [inline website/docs/r/cosmosdb_account.html.markdown]: Some context: the Azure API's won't allow this field to be set to any value unless `consstency_level` is set to `BoundedStaleness`

I'm unsure if that's an API bug, or intentional design decision - that's my only concern with this PR AFAIK

**Review by grubernaut — APPROVED** [review]
LGTM

File: website/docs/r/cosmosdb_account.html.markdown (hunk 1 above)
Comment (stack72) [inline website/docs/r/cosmosdb_account.html.markdown]: let's just make that a NOTE in the docs

**Pull request merged**